compatibility/bazel_tools/daml.cc.tpl

```diff
@@ -0,0 +1,87 @@
+#include "tools/cpp/runfiles/runfiles.h"
+#include <codecvt>
+#include <iostream>
+#include <locale>
+#include <memory>
+#include <unistd.h>
+#if defined(_WIN32)
+#include <windows.h>
+#endif
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+/*
+NOTE (MK): Why is this not just an sh_binary?
+Good question! It turns out that on Windows, an
+sh_binary creates a custom executable that then launches
+bash. Somehow this seems to result in a double fork
+(or the Windows-equivalent) thereof which results in
+child processes not being childs of the original process.
+This makes it really hard to kill a process including
+all child processes which is really important since
+otherwise you are left with a running JVM process
+if you kill `daml sandbox`.
+*/
+
+int main(int argc, char **argv) {
+    std::string error;
+    std::unique_ptr<Runfiles> runfiles(Runfiles::Create(argv[0], &error));
+    if (runfiles == nullptr) {
+        std::cerr << "Failed to initialize runfiles: " << error << "\n";
+        exit(1);
+    }
+    const char *java_home = getenv("JAVA_HOME");
+    std::string path_prefix = "";
+    // NOTE (MK) I don’t entirely understand when Bazel defines
+    // JAVA_HOME and when it doesn’t but it looks like it defines
+    // it in tests but not in genrules which is good enough for us.
```

Comment:
We're setting `build --test_env=JAVA_HOME` in `.bazelrc` which makes `JAVA_HOME` exported in tests. For `genrule`s you could probably also use the [java make variables](https://docs.bazel.build/versions/master/be/make-variables.html#custom_variables) that Bazel offers.

Reply:
Ah thanks for the explanation! I don’t actually need it in genrules, I mostly noticed that it wasn’t set there when my C++ code segfaulted when I didn’t handle the case where `JAVA_HOME` wasn’t set :slightly_smiling_face: